### PR TITLE
Feature/plots

### PR DESCRIPTION
**Pull request by Totoketchup**

### CODE REVIEW

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/nodeInformation.jsx [hunk 1]
```diff
@@ -43,6 +46,14 @@ const NodeInformation = ({ closeNodeInformation, selectedNodePath, tree }) => {
     const treeVersion = semver.major(tree._version);
     const treeData = tree.trees[Object.keys(tree.trees)[0]];
     const selectedNode = findSelectedNode(selectedNodePath, treeData);
+    let min = undefined;
+    let max = undefined;
+    console.log(interpreter);
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/nodeInformation.jsx]: remove

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 2]
```diff
@@ -8,22 +12,54 @@ const Ul = styled('ul')`
...
+            distribution={ value }
+            outputValues={ outputValues }
+            size={ size }
+          ></Histogram>
+        </div>;
+      }
+      else {
+        return <div>
+          {samplesIt}
+          <BoxPlot
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]:
```
<BoxPlot
  ... />
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 3]
```diff
@@ -8,22 +12,54 @@ const Ul = styled('ul')`
...
+          {samplesIt}
+          <Histogram
+            distribution={ value }
+            outputValues={ outputValues }
+            size={ size }
+          ></Histogram>
+        </div>;
+      }
+      else {
+        return <div>
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]:
```
return (
  <div>
    ...
   </div>
);
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 4]
```diff
@@ -8,22 +12,54 @@ const Ul = styled('ul')`
...
-  if (node.prediction) {
+const Statistics = ({ node, totalMin, totalMax, treeVersion, outputValues }) => {
+  if (treeVersion == 2) {
+    const plot = () => {
+      const { value, standard_deviation, min, max, size } = interpreter.distribution(node);
+      const samplesIt = <Ul>
+        <li>{size} samples</li>
+      </Ul>;
+      if (_.isUndefined(standard_deviation)) {
+        return <div>
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]:
```
return (
  <div>
    ...
   </div>
);
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 5]
```diff
@@ -8,22 +12,54 @@ const Ul = styled('ul')`
...
+          {samplesIt}
+          <Histogram
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]:
```
<Histogram
  .... />
```

File: packages/react-craft-ai-decision-tree/stories/decisionTree.js [hunk 6]
```diff
@@ -112,16 +113,29 @@ storiesOf('Tree displayed with fixed height', module)
       <DecisionTreeWithPanel data={ tree } />
     </div>
   ))
-  .add('display tree v2', () => (
+  .add('display tree v2 regression', () => (
     <div
       style={{
         display: 'flex',
-        height: 500,
+        height: '100vh',
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/stories/decisionTree.js]: use px

File: packages/react-craft-ai-decision-tree/stories/decisionTree.js [hunk 7]
```diff
@@ -112,16 +113,29 @@ storiesOf('Tree displayed with fixed height', module)
...
+      <div style={{ height: 'inherit', flexGrow: 1 }}>
+        <DecisionTreeWithPanel data={ treeV2Reg } />
+      </div>
+    </div>
+  ))
+  .add('display tree v2 classification', () => (
+    <div
+      style={{
+        display: 'flex',
+        height: '100vh',
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/stories/decisionTree.js]: use px

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/boxplot.jsx [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    svg
+      .append('text')
+      .attr('class', localTextCssClass)
+      .attr('y', (height - stdRectangleHeight) / 2 - 10)
+      .attr('x', scaleX(mean))
+      .text((Math.round(mean * 100) / 100));
+  }
+
+  render() {
+    return <svg ref={ (node) => this.node = node }></svg>;
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/boxplot.jsx]:
```
return (
  <svg ... />
);
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+  border-radius: 8px;			
+  pointer-events: none;
+  transform: translateX(-50%);
+  padding: 1px 10px 1px 10px;
+  white-space: nowrap;
+`;
+
+
+class Histogram extends React.Component {
+  // Draw here svg components that are static and have therefore to be drawn only once
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx]: what are you "static" component

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+          .style('opacity', 0);
+      });
+    
+    rectback
+      .exit()
+      .remove();
+  }
+
+  render() {
+    return <div ref={ (div) => this.div = div } style={{ position: 'relative' }}>
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx]:
```
return (
  <div ...>
     ...
  </div>
);
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+      <svg ref={ (node) => this.node = node }>
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx]:
```
<svg ... />
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 12]
```diff
@@ -8,22 +12,57 @@ const Ul = styled('ul')`
   padding-inline-start: 0;
 `;
 
-const Statistics = ({ node }) => {
-  if (node.prediction) {
+const Statistics = ({ node, totalMin, totalMax, treeVersion, outputValues }) => {
+  if (treeVersion == 2) {
+    const plot = () => {
+      const { value, standard_deviation, min, max, size } = interpreter.distribution(node);
+      const samplesIt = <Ul>
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]:
```
 const samplesIt = (
  <Ul>
    <li>{size} samples</li>
  </Ul>
);
```

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx [hunk 13]
```diff
@@ -8,22 +12,57 @@ const Ul = styled('ul')`
...
+        );
+      }
+    };
     return (
       <div className='node-predictions'>
         <H3NodeInformation>Statistics</H3NodeInformation>
-        <Ul>
-          <li>{node.prediction.nb_samples} samples</li>
-        </Ul>
+        {plot()}
```
Comment (Giska) [inline packages/react-craft-ai-decision-tree/src/components/nodeInformation/statistics.jsx]: we need to talk about that

File: packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx (hunk 9 above)
Comment (Totoketchup) [reply packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx]: I just want the axes to be drawn just once but maybe there is a clever way than doing this

Comment (Totoketchup) [reply packages/react-craft-ai-decision-tree/src/components/nodeInformation/histogram.jsx]: ok it's corrected :)

**Pull request merged**